### ISSUE TITLE
check if slash failure received by oracle consumer chain

### ISSUE THREAD

**Issue by oraclown** [tellor-io/tellor-parachain-contracts#67]
A dispute is started on the oracle consumer parachain, then `beginParachainDispute` is called on the evm-compatible parachain. If the disputed reporter (staker) being slashed has zero balance, `beginParachainDispute` will revert since `slashParachainReporter` fails with `zero staker balance` message.

Is this case accounted for on the pallet side? So that dispute won't open (in this case) and storage is cleaned up?

**Comment by oraclown** [comment]
We may have already checked this...

**Comment by oraclown** [comment]
> I propose that we just remove the "require bal > 0" check, and make sure that we correctly handle a slash amount > 0 on the EVM-chain side. Justification: we want to prioritize being able to dispute bad values

So on the contracts side, just remove the require and don't attempt the transfer if disputed reporter has zero balance to begin with?

**Comment by tkernell** [comment]
Yes exactly. This is a good catch (and would improve eth mainnet tellor as well, even though this case should never come up in theory)

**Comment by evilrobot-01** [comment]
No action is currently taken as the pallet shouldnt be able to slash a reporter without being instructed to do so from the governance contract. Put another way, https://github.com/tellor-io/governance/blob/0dcc2ad501b1e51383a99a22c60eeb8c36d61bc3/contracts/Governance.sol#L150 does not exist in the pallet implementation at https://github.com/tellor-io/tellor-pallet/blob/f5bf011eca27cb19015c52d57d3f60734b7bc8a3/src/lib.rs#L1202-L1213.

Shall I create an issue to ensure that the disputed reporter stake amount >= slash amount within the begin_dispute function on the pallet? It seems wasteful to begin a dispute, send a cross chain message, only for it to fail.

**Comment by evilrobot-01** [comment]
I guess the dispute initiator wouldn't bother disputing a value if there was no reporter stake amount, greater than the dispute fee (or equivalent value), which they could win.

I don't believe there is anything more to do on the pallet side, but will reflect and review in the morning.

**Comment by tkernell** [comment]
I would imagine that maybe a creator or user of a protocol which depends on this data might still be incentivized to dispute bad data, either out of goodwill, desire to see their protocol consume only good data, or financial self-interest as a consequence of the protocol's consumption of the data.
